axon-mongo-spring-boot-autoconfigure/src/test/java/org/axonframework/extensions/mongo/springboot/autoconfig/MongoAutoConfigurationDefaultTest.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2010-2022. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.extensions.mongo.springboot.autoconfig;
+
+import com.mongodb.ClientSessionOptions;
+import com.mongodb.client.ClientSession;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import org.axonframework.common.transaction.TransactionManager;
+import org.axonframework.eventhandling.tokenstore.TokenStore;
+import org.axonframework.eventsourcing.eventstore.EventStorageEngine;
+import org.axonframework.modelling.saga.repository.SagaStore;
+import org.axonframework.serialization.Serializer;
+import org.bson.Document;
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.api.extension.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.autoconfigure.data.mongo.MongoDataAutoConfiguration;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.EnableMBeanExport;
+import org.springframework.data.mongodb.MongoDatabaseFactory;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.convert.MappingMongoConverter;
+import org.springframework.jmx.support.RegistrationPolicy;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration
+@EnableAutoConfiguration(exclude = {
+        MongoDataAutoConfiguration.class
+})
+//@TestPropertySource("classpath:test.metrics.application.properties")
+@EnableMBeanExport(registration = RegistrationPolicy.IGNORE_EXISTING)
+class MongoAutoConfigurationDefaultTest {
+
+    @Autowired
+    private ApplicationContext applicationContext;
+
+    @Test
+    void contextInitialization() {
```

Comment:
It's a bit of a flavor thing perhaps, but I believe the internal Spring tests around boot use the `ApplicationContextRunner` for testing bean creation and the likes.
We have a mix in the AF repository, because the `ApplicationContextRunner` was introduced after the initial introducing of the Spring Boot integration.
The `XStreamAutoConfigurationTest` is an example of the new approach, by the way.

Reply:
The fact I'm sharing it, is because it allows easier adjustment of the application context for testing. Hence, the necessity of a test class for the default and for the one using properties blurs quite some.

Reply:
Furthermore, it opens a door to adding a test case that uses these new Mongo Starters and the Framework starters. 
That way, we can clearly define the behavior of which stores turn up when users have a combination.

Granted, I am not a 100% convinced this project should fulfill that purpose...then again, it is intended to extend AxonFramework, so the chances are high both starters are used.

Reply:
I'm not sure we indeed can clearly define the behavior, as it depends a lot on specific configuration and what else is on the class path.

Reply:
Naah, agreed. I think how you've changed the test cases right now is awesome. So let's leave this point for the future, once we need something like this.